docs/API.md

```diff
@@ -31,35 +31,161 @@ The following tags are available to override Datadog specific options:
 
 <h3 id="scope-manager">Scope Manager</h3>
 
-In order to provide context propagation, this library includes a scope manager. A scope is basically a wrapper around a span that can cross both synchronous and asynchronous contexts.
+In order to provide context propagation, this library includes a scope manager.
+A scope is basically a wrapper around a span that can cross both synchronous and
+asynchronous contexts.
 
-For example:
+The scope manager contains 3 APIs available on `tracer.scope()`:
+
+<h4>scope.active()</h4>
+
+This method returns the active span from the current scope.
+
+<h4>scope.activate(span, fn)</h4>
+
+This method activates the provided span in a new scope available in the function
+provided as the second argument. Any asynchronous context created from whithin
+that function will also have the same scope.
 
 ```javascript
-const tracer = require('dd-trace').init({ plugins: false })
-const express = require('express')
-const app = express()
+const tracer = require('dd-trace').init()
+const scope = tracer.scope()
+const span = tracer.startSpan('web.request')
+const promise = Promise.resolve()
+const log = console.log
 
-app.use((req, res, next) => {
-  const span = tracer.startSpan('web.request')
-  
-  tracer.scope().activate(span, () => next())
-})
+scope.activate(span, () => {
+  log(scope.active()) // the span because in new scope
+
+  someFunction() // the span because called in scope
 
-app.get('/hello', (req, res, next) => {
   setTimeout(() => {
-    const span = tracer.scope().active() // the span activated earlier
+    log(scope.active()) // the span because setTimeout called in scope
+  })
+
+  promise.then(() => {
+    log(scope.active()) // the span because then() called in scope
```

Comment:
I almost wonder if `scope` is confusing because, the scope of the function vs the scope of the span

Reply:
Just a random thought, no prescribed changes

Reply:
There is no distinction. The span is always in the scope of the function. Unless I'm misunderstanding what you mean?

Reply:
> called in scope

the scope of what, the js scope or the tracing scope?

Like I said, it is weird/confusing in general, but just a random thought, not suggesting any specific change here.